src/server.cpp

```diff
@@ -20,19 +23,55 @@ void Server::start_receive()
         std::bind(&Server::handle_receive, this, pls::_1, pls::_2));
 }
 
-void Server::handle_receive(const error_code& error, std::size_t /*bytes_transferred*/)
+void Server::handle_receive(const error_code& error, std::size_t bytes)
 {
-    if (!error || error == boost::asio::error::message_size)
+    if (!error)
     {
-        const std::string message("Hello, World!");
+        size_t length;
+        try
+        {
+            length = recv_buffer_.at(2) * 256 + recv_buffer_.at(3);
+        }
+        catch (std::out_of_range const& exc)
+        {
+            std::cout << exc.what() << "\n";
+        }
 
-        socket_.async_send_to(boost::asio::buffer(message), remote_endpoint_,
-          std::bind(&Server::handle_send, this, message,pls::_1, pls::_2));
+        if (bytes >= length)
+        {
+            uint8_t send_code;
 
-        start_receive();
+            if (recv_buffer_[0] == ACCESS_REQUEST)
+                send_code = ACCESS_ACCEPT;
+
+            send_buffer_[0] = send_code;
+            send_buffer_[1] = recv_buffer_[1];
+            send_buffer_[2] = 0;
+            send_buffer_[3] = 20;
+
+            for (int i = 4; i <= 19; i++)
+                send_buffer_[i] = recv_buffer_[i];
```

Comment:
I propose you to rewrite this cycle as the following:
```c++
for (size_t i = 0; i < 20; ++i)
    send_buffer_[i + 4] = recv_buffer_[i + 4];
```

The reasons are:
 * use `size_t` instead of `int` shows that you work with sizes, not some arbitrary values (and `size_t` is unsigned, a size cannot be negative);
 * start indexing from 0 to some size shows that you work with some value, not an array span;
 * it is always nice to use strict comparisons - `<` and `>` instead of `<=` and `>=`;
 * `++i` is not better that `i++` in this case, but in general you should preffer this form;
 * offset 4 shows you the position of the value you are working with in the buffer.

Reply:
Besides, why 20 (19 in your case)? I think you should use 16 (15 in your version of the code).